### PR TITLE
release 3.0.0

### PR DESCRIPTION
**Pull request by yuukiiwa**
## Major updates
- new DSL2 syntax
- Azure bug fix
- DNA structural variant detection
- RNA modification and fusion detection

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
    - [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/nanoseq/tree/master/.github/CONTRIBUTING.md)
    - [x] If necessary, also make a PR on the nf-core/nanoseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,71 +3,131 @@
...
+
+### Major enhancements
+
+- Add DNA variant calling functionality
+- Add RNA modification and fusion detection functionalities
+- Add `demux_fast5` module to output demultiplexed fast5 files when `--output_demultiplex_fast5` is set
+- Add `--trim_barcodes` in Guppy basecaller to trim the barcodes from output fastq
+- Port pipeline to the updated Nextflow DSL2 syntax adopted on nf-core/modules
+  - Removed `--publish_dir_mode` as it is no longer required for the new syntax
+- Bump minimum Nextflow version from 21.04.0 -> 21.10.3
```
Comment (maxulysse) [inline CHANGELOG.md:16]:
```suggestion
- Port pipeline to the updated Nextflow DSL2 syntax adopted on nf-core/modules
- Removed `--publish_dir_mode` as it is no longer required for the new syntax
- Bump minimum Nextflow version from 21.04.0 -> 21.10.3
```

File: CITATIONS.md [hunk 2]
```diff
@@ -10,80 +10,135 @@
...
+
+* [DeepVariant](https://pubmed.ncbi.nlm.nih.gov/30247488/)
+
+  > Poplin R, Chang PC, Alexander D, Schwartz S, Colthurst T, Ku A, Newburger D, Dijamco J, Nguyen N, Afshar PT, Gross SS, Dorfman L, McLean CY, DePristo MA. A universal SNP and small-indel variant caller using deep neural networks. Nat Biotechnol. 2018 Nov;36(10):983-987. doi: 10.1038/nbt.4235. Epub 2018 Sep 24. PMID: 30247488.
 
 * [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)
 
-* [featureCounts](https://www.ncbi.nlm.nih.gov/pubmed/24227677/)
-    > Liao Y, Smyth GK, Shi W. featureCounts: an efficient general purpose program for assigning sequence reads to genomic features. Bioinformatics. 2014 Apr 1;30(7):923-30. doi: 10.1093/bioinformatics/btt656. Epub 2013 Nov 13. PubMed PMID: 24227677.
+  > Quinlan AR, Hall IM. BEDTools: a flexible suite of utilities for comparing genomic features. Bioinformatics. 2010 Mar 15;26(6):841-2. doi: 10.1093/bioinformatics/btq033. Epub 2010 Jan 28. PubMed PMID: 20110278; PubMed Central PMCID: PMC2832824.
```
Comment (maxulysse) [inline CITATIONS.md:25]:
```suggestion
```

File: CITATIONS.md [hunk 3]
```diff
@@ -10,80 +10,135 @@
 
 ## Pipeline tools
 
-* [BEDTools](https://www.ncbi.nlm.nih.gov/pubmed/20110278/)
-    > Quinlan AR, Hall IM. BEDTools: a flexible suite of utilities for comparing genomic features. Bioinformatics. 2010 Mar 15;26(6):841-2. doi: 10.1093/bioinformatics/btq033. Epub 2010 Jan 28. PubMed PMID: 20110278; PubMed Central PMCID: PMC2832824.
+- [BEDTools](https://www.ncbi.nlm.nih.gov/pubmed/20110278/)
+
```
Comment (maxulysse) [inline CITATIONS.md:14]:
```suggestion

  > Quinlan AR, Hall IM. BEDTools: a flexible suite of utilities for comparing genomic features. Bioinformatics. 2010 Mar 15;26(6):841-2. doi: 10.1093/bioinformatics/btq033. Epub 2010 Jan 28. PubMed PMID: 20110278; PubMed Central PMCID: PMC2832824.
```

File: CITATIONS.md [hunk 4]
```diff
@@ -10,80 +10,135 @@
...
+* [cuteSV](https://pubmed.ncbi.nlm.nih.gov/32746918/)
+
+  > Jiang T, Liu Y, Jiang Y, Li J, Gao Y, Cui Z, Liu Y, Liu B, Wang Y. Long-read-based human genomic structural variation detection with cuteSV. Genome Biol. 2020 Aug 3;21(1):189. doi: 10.1186/s13059-020-02107-y. PMID: 32746918; PMCID: PMC7477834.
+
+* [DeepVariant](https://pubmed.ncbi.nlm.nih.gov/30247488/)
```
Comment (maxulysse) [inline CITATIONS.md:19]:
```suggestion
- [cuteSV](https://pubmed.ncbi.nlm.nih.gov/32746918/)

  > Jiang T, Liu Y, Jiang Y, Li J, Gao Y, Cui Z, Liu Y, Liu B, Wang Y. Long-read-based human genomic structural variation detection with cuteSV. Genome Biol. 2020 Aug 3;21(1):189. doi: 10.1186/s13059-020-02107-y. PMID: 32746918; PMCID: PMC7477834.

- [DeepVariant](https://pubmed.ncbi.nlm.nih.gov/30247488/)
```

File: CITATIONS.md [hunk 5]
```diff
@@ -10,80 +10,135 @@
...
+
+  > Poplin R, Chang PC, Alexander D, Schwartz S, Colthurst T, Ku A, Newburger D, Dijamco J, Nguyen N, Afshar PT, Gross SS, Dorfman L, McLean CY, DePristo MA. A universal SNP and small-indel variant caller using deep neural networks. Nat Biotechnol. 2018 Nov;36(10):983-987. doi: 10.1038/nbt.4235. Epub 2018 Sep 24. PMID: 30247488.
 
 * [FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/)
```
Comment (maxulysse) [inline CITATIONS.md:24]:
```suggestion
```

File: CITATIONS.md [hunk 6]
```diff
@@ -10,80 +10,135 @@
...
 
-* [NanoLyse](https://pubmed.ncbi.nlm.nih.gov/29547981/)
-    > De Coster, W., D’Hert, S., Schultz, D. T., Cruts, M., & Van Broeckhoven, C. (2018). NanoPack: visualizing and processing long-read sequencing data. Bioinformatics, 34(15), 2666-2669. PubMed PMID: 29547981; PubMed Central PMCID: PMC6061794.
+* [PEPPER-Margin-DeepVariant](https://pubmed.ncbi.nlm.nih.gov/34725481/)
 
-* [NanoPlot](https://pubmed.ncbi.nlm.nih.gov/29547981/)
-    > De Coster W, D'Hert S, Schultz DT, Cruts M, Van Broeckhoven C. NanoPack: visualizing and processing long-read sequencing data. Bioinformatics. 2018 Aug 1;34(15):2666-2669. doi: 10.1093/bioinformatics/bty149. PubMed PMID: 29547981; PubMed Central PMCID: PMC6061794.
+  > Shafin K, Pesout T, Chang PC, Nattestad M, Kolesnikov A, Goel S, Baid G, Kolmogorov M, Eizenga JM, Miga KH, Carnevali P, Jain M, Carroll A, Paten B. Haplotype-aware variant calling with PEPPER-Margin-DeepVariant enables high accuracy in nanopore long-reads. Nat Methods. 2021 Nov;18(11):1322-1332. doi: 10.1038/s41592-021-01299-w. Epub 2021 Nov 1. PMID: 34725481; PMCID: PMC8571015.
 
 * [pycoQC](https://doi.org/10.21105/joss.01236)
```
Comment (maxulysse) [inline CITATIONS.md:51]:
```suggestion
```

File: CITATIONS.md [hunk 7]
```diff
@@ -10,80 +10,135 @@
...
-    > Leger A, Leonardi T, (2019). pycoQC, interactive quality control for Oxford Nanopore Sequencing. Journal of Open Source Software, 4(34), 1236.
 
-* [qcat](https://github.com/nanoporetech/qcat)
+  > Leger A, Leonardi T, (2019). pycoQC, interactive quality control for Oxford Nanopore Sequencing. Journal of Open Source Software, 4(34), 1236.
+
```
Comment (maxulysse) [inline CITATIONS.md:54]:
```suggestion
```

File: README.md [hunk 8]
```diff
@@ -86,9 +96,9 @@ An example input samplesheet for performing both basecalling and demultiplexing
...
 
-Many thanks to others who have helped out along the way too, including (but not limited to): [@crickbabs](https://github.com/crickbabs), [@AnnaSyme](https://github.com/AnnaSyme).
+Many thanks to others who have helped out along the way too, including (but not limited to): [@crickbabs](https://github.com/crickbabs), [@AnnaSyme](https://github.com/AnnaSyme),[@ekushele](https://github.com/ekushele).
```
Comment (maxulysse) [inline README.md:101]:
```suggestion
Many thanks to others who have helped out along the way too, including (but not limited to): [@crickbabs](https://github.com/crickbabs), [@AnnaSyme](https://github.com/AnnaSyme), [@ekushele](https://github.com/ekushele).
```

File: nextflow.config [hunk 9]
```diff
@@ -174,17 +180,37 @@ profiles {
...
+} else {
+    params.genomes = [:]
+}
+
+// Load igenomes.config if required
+if (!params.igenomes_ignore) {
+    includeConfig 'conf/igenomes.config'
+} else {
+    params.genomes = [:]
 }
```
Comment (maxulysse) [inline nextflow.config:203]:
```suggestion
```

Duplicate code

File: nextflow.config [hunk 10]
```diff
@@ -214,10 +240,13 @@ manifest {
     homePage        = 'https://github.com/nf-core/nanoseq'
     description     = 'A pipeline to demultiplex, QC and map Nanopore data'
     mainScript      = 'main.nf'
-    nextflowVersion = '!>=21.04.0'
+    nextflowVersion = '!>=21.10.3'
     version         = '2.0.1'
```
Comment (maxulysse) [inline nextflow.config:244]:
```suggestion
    version         = '3.0.0'
```
Was `nf-core bump version 3.0.0` run?

File: modules/local/bam_rename.nf [hunk 11]
```diff
@@ -1,14 +1,10 @@
...
-        container "https://containers.biocontainers.pro/s3/SingImgsRepo/biocontainers/v1.2.0_cv1/biocontainers_v1.2.0_cv1.img"
-    } else {
-        container "biocontainers/biocontainers:v1.2.0_cv1"
-    }
+    //container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+    //    'https://containers.biocontainers.pro/s3/SingImgsRepo/biocontainers/v1.2.0_cv1/biocontainers_v1.2.0_cv1.img' :
+    //    'quay.io/biocontainers/biocontainers:latest' }"
 
     input:
     tuple val(meta), path(bam)
```
[the remark this answers is not in the document]
Comment (yuukiiwa) [reply modules/local/bam_rename.nf:10]: @drpatelh came up with this around two years ago to solve the bam input problem

File: README.md [hunk 12]
```diff
@@ -1,67 +1,77 @@
-# ![nfcore/nanoseq](docs/images/nf-core-nanoseq_logo.png)
+# ![nf-core/nanoseq](docs/images/nf-core-nanoseq_logo_light.png#gh-light-mode-only) ![nf-core/nanoseq](docs/images/nf-core-nanoseq_logo_dark.png#gh-dark-mode-only)
 
 [![GitHub Actions CI Status](https://github.com/nf-core/nanoseq/workflows/nf-core%20CI/badge.svg)](https://github.com/nf-core/nanoseq/actions?query=workflow%3A%22nf-core+CI%22)
 [![GitHub Actions Linting Status](https://github.com/nf-core/nanoseq/workflows/nf-core%20linting/badge.svg)](https://github.com/nf-core/nanoseq/actions?query=workflow%3A%22nf-core+linting%22)
-[![AWS CI](https://img.shields.io/badge/CI%20tests-Full%20size-FF9900?labelColor=000000&logo=Amazon%20AWS)](https://nf-co.re/nanoseq/results)
-[![Cite with Zenodo](http://img.shields.io/badge/DOI-10.5281/zenodo.3697959-1073c8?labelColor=000000)](https://doi.org/10.5281/zenodo.3697959)
+[![AWS CI](https://img.shields.io/badge/CI%20tests-full%20size-FF9900?logo=Amazon%20AWS)](https://nf-co.re/nanoseq/results)
+[![Cite with Zenodo](http://img.shields.io/badge/DOI-10.5281/zenodo.XXXXXXX-1073c8)](https://doi.org/10.5281/zenodo.XXXXXXX)
```
Comment (maxulysse) [inline README.md:6]:
```suggestion
[![Cite with Zenodo](http://img.shields.io/badge/DOI-10.5281/zenodo.3697959-1073c8)](https://doi.org/10.5281/zenodo.3697959)
```

File: bin/create_yml.py [hunk 13]
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env python
+
```
Comment (maxulysse) [inline bin/create_yml.py:2]: can we get more info about this script?

Comment (yuukiiwa) [reply bin/create_yml.py:2]: This script is for the differential RNA modification analysis using `xpore`, which requires a `config.yml` specifying the two sample conditions as input

File: assets/multiqc_config.yaml [hunk 14]
```diff
@@ -1,11 +1,11 @@
 report_comment: >
```
Comment (maxulysse) [inline assets/multiqc_config.yaml:1]: You shouldn't need this file anymore since you have the `assets/multiqc_config.yml`

Comment (yuukiiwa) [reply assets/multiqc_config.yaml:1]: I have deleted this in the PR from the `variant_calling` branch to `dev`:
https://github.com/nf-core/nanoseq/pull/185
Thank you!!

**Review by maxulysse — CHANGES_REQUESTED** [review]
Can you update `lib/WorkflowMain.groovy` with the zenodo DOI for the nanoseq pipeline?
it has the one from the rnaseq one.

I feel that some local modules can be replaced by some nf-core modules, and other local modules can be pushed to nf-core/modules, but that's not really blocking for a release, that can be addressed in a following release.

Did not find any other issues that the ones that were already raised and fixed.

**Comment by yuukiiwa** [comment]
Thank you so much for reviewing, @maxulysse! 

I have updated the `lib/WorkflowMain.groovy` with `nanoseq`'s zenodo DOI in PR https://github.com/nf-core/nanoseq/pull/186.

I have discussed with @christopher-hakkaart (and a year ago with @drpatelh) on the feasibility of using the shared `nf-core/modules` for `nanoseq`, which is designed differently from other nf-core pipelines as `nanoseq` allows each sample to have its own reference `fasta` and/or `gtf` file(s) while most nf-core pipelines take in one reference `fasta` and/or `gtf` file(s) for all samples. This makes the channeling of reference files in `nanoseq` different from that in other nf-core pipelines.

File: CITATIONS.md [hunk 15]
```diff
@@ -10,80 +10,127 @@
...
-* [qcat](https://github.com/nanoporetech/qcat)
+- [Guppy](https://nanoporetech.com/nanopore-sequencing-data-analysis)
 
-* [SAMtools](https://www.ncbi.nlm.nih.gov/pubmed/19505943/)
-    > Li H, Handsaker B, Wysoker A, Fennell T, Ruan J, Homer N, Marth G, Abecasis G, Durbin R; 1000 Genome Project Data Processing Subgroup. The Sequence Alignment/Map format and SAMtools. Bioinformatics. 2009 Aug 15;25(16):2078-9. doi: 10.1093/bioinformatics/btp352. Epub 2009 Jun 8. PubMed PMID: 19505943; PubMed Central PMCID: PMC2723002.
+- [JAFFAL](https://doi.org/10.1186/s13059-021-02588-5)
 
-* [StringTie2](https://www.ncbi.nlm.nih.gov/pubmed/31842956/)
-    > Kovaka S, Zimin AV, Pertea GM, Razaghi R, Salzberg SL, Pertea M. Transcriptome assembly from long-read RNA-seq alignments with StringTie2 Genome Biol. 2019 Dec 16;20(1):278. doi: 10.1186/s13059-019-1910-1. PubMed PMID: 31842956; PubMed Central PMCID: PMC6912988.
+  > Davidson NM, et al., JAFFAL: detecting fusion genes with long-read transcriptome sequencing. Genome Biology (2022)
```
Comment (christopher-hakkaart) [inline CITATIONS.md:37]:
```suggestion
  > Davidson NM, Chen Y, Sadras T, Ryland GL, Blombery P, Ekert PG, Göke J, Oshlack A. JAFFAL: detecting fusion genes with long-read transcriptome sequencing. Genome Biol. 2022 Jan 6;23(1):10. doi: 10.1186/s13059-021-02588-5. PMID: 34991664; PMCID: PMC8739696.
```

File: CITATIONS.md [hunk 16]
```diff
@@ -10,80 +10,127 @@
...
+- [StringTie2](https://www.ncbi.nlm.nih.gov/pubmed/31842956/)
+
+  > Kovaka S, Zimin AV, Pertea GM, Razaghi R, Salzberg SL, Pertea M. Transcriptome assembly from long-read RNA-seq alignments with StringTie2 Genome Biol. 2019 Dec 16;20(1):278. doi: 10.1186/s13059-019-1910-1. PubMed PMID: 31842956; PubMed Central PMCID: PMC6912988.
+
+- [UCSC tools](https://www.ncbi.nlm.nih.gov/pubmed/20639541/)
+
+  > Kent WJ, Zweig AS, Barber G, Hinrichs AS, Karolchik D. BigWig and BigBed: enabling browsing of large distributed datasets. Bioinformatics. 2010 Sep 1;26(17):2204-7. doi: 10.1093/bioinformatics/btq351. Epub 2010 Jul 17. PubMed PMID: 20639541; PubMed Central PMCID: PMC2922891.
+
+- [xPore](https://doi.org/10.1038/s41587-021-00949-w)
+  > Pratanwanich PN, et al.,Identification of differential RNA modifications from nanopore direct RNA sequencing with xPore. Nat Biotechnol (2021)
```
Comment (christopher-hakkaart) [inline CITATIONS.md:88]:
```suggestion
  > Pratanwanich PN, Yao F, Chen Y, Koh CWQ, Wan YK, Hendra C, Poon P, Goh YT, Yap PML, Chooi JY, Chng WJ, Ng SB, Thiery A, Goh WSS, Göke J. Identification of differential RNA modifications from nanopore direct RNA sequencing with xPore. Nat Biotechnol. 2021 Nov;39(11):1394-1402. doi: 10.1038/s41587-021-00949-w. Epub 2021 Jul 19. PMID: 34282325.
```

File: README.md [hunk 17]
```diff
@@ -1,67 +1,77 @@
...
-2. Present QC for raw reads ([`MultiQC`](http://multiqc.info/))
-
 On release, automated continuous integration tests run the pipeline on a [full-sized dataset](https://github.com/nf-core/test-datasets/tree/nanoseq#full-sized-test-data) obtained from the [Singapore Nanopore Expression Consortium](https://github.com/GoekeLab/sg-nex-data) on the AWS cloud infrastructure. This ensures that the pipeline runs on AWS, has sensible resource allocation defaults set to run on real-world datasets, and permits the persistent storage of results to benchmark between pipeline releases and other analysis sources. The results obtained from the full-sized test can be viewed on the [nf-core website](https://nf-co.re/nanoseq/results).
 
 ## Pipeline Summary
 
-1. Basecalling and/or demultiplexing ([`Guppy`](https://nanoporetech.com/nanopore-sequencing-data-analysis) or [`qcat`](https://github.com/nanoporetech/qcat); *optional*)
+1. Basecalling and/or demultiplexing ([`Guppy`](https://nanoporetech.com/nanopore-sequencing-data-analysis), [`demux_fast5`](https://github.com/nanoporetech/ont_fast5_api#demux_fast5) or [`qcat`](https://github.com/nanoporetech/qcat); _optional_)
```
Comment (christopher-hakkaart) [inline README.md:28]:
```suggestion
1. Basecalling and/or demultiplexing ([`Guppy`](https://nanoporetech.com/nanopore-sequencing-data-analysis), [`demux_fast5`](https://github.com/nanoporetech/ont_fast5_api#demux_fast5), or [`qcat`](https://github.com/nanoporetech/qcat); _optional_)
```

File: CITATIONS.md (hunk 15 above)
Comment (yuukiiwa) [reply CITATIONS.md:37]: I have changed this in PR #186

File: CITATIONS.md (hunk 16 above)
Comment (yuukiiwa) [reply CITATIONS.md:88]: I have changed this in PR https://github.com/nf-core/nanoseq/pull/186

File: README.md (hunk 17 above)
Comment (yuukiiwa) [reply README.md:28]: I have changed this in PR https://github.com/nf-core/nanoseq/pull/186

**Review by christopher-hakkaart — CHANGES_REQUESTED** [review]
I powered through this quickly so I hope I didn't miss anything too obvious. I  found a few little things which I have added suggestions for.

I don't see much holding back the new release although I think there are a small number of things that could be tidied quickly for a 3.1 release:
- Docker/singularity images could be moved to public repos
- Local modules should have extra.args added to make more flexible
- Rename bams could be moved to a process.

In the long term, I like the idea of moving local/modules > nf-core/modules. It would be tricky with nanoseq carrying around its own genome, which contrasts with other pipelines. It will take a bit more thinking about how we could do this to share modules and eventually sub-workflows with other pipelines. The bottom line is that if we can move in the same ways as the community then we don't have to maintain as much - which would be fantastic!

**Review by maxulysse — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 18]
```diff
@@ -8,47 +8,60 @@ on:
...
     strategy:
       matrix:
-        profiles: [test_bc_nodx, test_nobc_dx, test_nobc_nodx, test_nobc_nodx_noaln]
+        profiles:
+          - "test_bc_nodx"
+          - "test_nobc_dx"
+          - "test_nobc_nodx_vc"
+          - "test_nobc_nodx_stringtie"
+          - "test_nobc_nodx_noaln"
+          - "test_nobc_nodx_rnamod"
```
[the remark this answers is not in the document]
Comment (yuukiiwa) [reply .github/workflows/ci.yml:64]: `- "test_nobc_nodx_vc"` is included in line 61

File: CITATIONS.md [hunk 19]
```diff
@@ -10,80 +10,127 @@
...
+
+  > De Coster, W., D’Hert, S., Schultz, D. T., Cruts, M., & Van Broeckhoven, C. (2018). NanoPack: visualizing and processing long-read sequencing data. Bioinformatics, 34(15), 2666-2669. PubMed PMID: 29547981; PubMed Central PMCID: PMC6061794.
+
+- [NanoPlot](https://pubmed.ncbi.nlm.nih.gov/29547981/)
+
+  > De Coster W, D'Hert S, Schultz DT, Cruts M, Van Broeckhoven C. NanoPack: visualizing and processing long-read sequencing data. Bioinformatics. 2018 Aug 1;34(15):2666-2669. doi: 10.1093/bioinformatics/bty149. PubMed PMID: 29547981; PubMed Central PMCID: PMC6061794.
+
+- [pycoQC](https://doi.org/10.21105/joss.01236)
+
+  > Leger A, Leonardi T, (2019). pycoQC, interactive quality control for Oxford Nanopore Sequencing. Journal of Open Source Software, 4(34), 1236.
```
[the remark this answers is not in the document]
Comment (yuukiiwa) [reply CITATIONS.md:67]: I have updated this

File: docs/output.md [hunk 20]
```diff
@@ -4,83 +4,74 @@
 
 ## Introduction
 
-This document describes the output produced by the pipeline. Most of the plots are taken from the MultiQC report, which summarises results at the end of the pipeline.
+This document describes the output produced by the nf-core/nanoseq pipeline. Most of the plots shown here are taken from the MultiQC report, which summarises the pipeline results.
 
 ## Pipeline overview
 
-The pipeline is built using [Nextflow](https://www.nextflow.io/). See [`main README.md`](../README.md) for a condensed overview of the steps in the pipeline, and the bioinformatics tools used at each step.
+The nf-core/nanoseq pipeline is built using [Nextflow](https://www.nextflow.io/). There are many different potential outputs for the pipeline depending on what file inputs and parameters you use. Please see [`main README.md`](../README.md) for a condensed overview of the steps in the pipeline, and the bioinformatics tools used at each step.
```
[the remark this answers is not in the document]
Comment (yuukiiwa) [reply docs/output.md:11]: I have updated this

File: docs/output.md [hunk 21]
```diff
@@ -4,83 +4,74 @@
...
 </details>
 
-*Documentation*:
+_Documentation_:
 [NanoLyse](https://github.com/wdecoster/nanolyse)
 
-*Description*:
-If you would like to run NanoLyse on the raw FastQ files then you can provide `--run_nanolyse` when running the pipeline. By default, the pipeline will filter the raw reads relative to lambda phage but you can provide your own fasta file of "contaminants" with `--nanolyse_fasta`. The filtered FastQ files will contain raw reads without the provided reference sequences (default: lambda phage sequences).
+_Description_:
+If you would like to run NanoLyse on the raw FASTQ files you can provide `--run_nanolyse` when running the pipeline. By default, the pipeline will filter lambda phage reads, however you can provide your own FASTA file of "contaminants" with `--nanolyse_fasta`. The filtered FASTQ files will contain raw reads without the specified reference sequences (default: lambda phage sequences).
```
[the remark this answers is not in the document]
Comment (yuukiiwa) [reply docs/output.md:57]: I have updated this

File: docs/output.md [hunk 22]
```diff
@@ -4,83 +4,74 @@
...
 </details>
 
-*Documentation*:
+_Documentation_:
 [PycoQC](https://github.com/a-slide/pycoQC), [NanoPlot](https://github.com/wdecoster/NanoPlot)
 
-*Description*:
-*PycoQC* and *NanoPlot* compute metrics and generate QC plots using the sequencing summary information generated by *Guppy* e.g. distribution of read length, read length over time, number of reads per barcode and other general stats. *NanoPlot* also generates QC metrics directly from fastq files as described in the next section.
+_Description_:
+_PycoQC_ and _NanoPlot_ can compute QC metrics and generate plots using the sequencing summary information generated by _Guppy_ e.g. distribution of read length, read length over time, number of reads per barcode and other general stats. _NanoPlot_ can also generates QC metrics directly from FASTQ files as described in the next section.
```
[the remark this answers is not in the document]
Comment (yuukiiwa) [reply docs/output.md:74]: I have updated this

File: docs/output.md [hunk 23]
```diff
@@ -89,68 +80,106 @@ If you would like to run NanoLyse on the raw FastQ files then you can provide `-
...
 </details>
 
-*Documentation*:
+_Documentation_:
 [NanoPlot](https://github.com/wdecoster/NanoPlot), [FastQC](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/)
 
-*Description*:
-*NanoPlot* can also be used to produce general quality metrics from the per-barcode fastq files generated by *Guppy* e.g. quality score distribution, read lengths and other general stats.
+_Description_:
+_NanoPlot_ can be used to produce general quality metrics from the per barcode FASTQ files generated by _Guppy_ e.g. quality score distribution, read lengths and other general stats.
```
[the remark this answers is not in the document]
Comment (yuukiiwa) [reply docs/output.md:93]: I have updated this

File: docs/output.md [hunk 24]
```diff
@@ -89,68 +80,106 @@ If you would like to run NanoLyse on the raw FastQ files then you can provide `-
...
 
 ![Nanoplot - Read quality vs read length](images/nanoplot_readlengthquality.png)
 
-*FastQC* gives general quality metrics about your reads. It provides information about the quality score distribution across your reads, the per base sequence content (%A/C/G/T). You get information about adapter contamination and other overrepresented sequences.
+_FastQC_ can give general quality metrics about your reads. It can provide information about the quality score distribution across your reads, the per base sequence content (%A/C/G/T). You can also generate information about adapter contamination and other over-represented sequences.
```
[the remark this answers is not in the document]
Comment (yuukiiwa) [reply docs/output.md:97]: I have updated this

File: docs/output.md [hunk 25]
```diff
@@ -89,68 +80,106 @@ If you would like to run NanoLyse on the raw FastQ files then you can provide `-
...
 
-*Description*:
-The [bigWig](https://genome.ucsc.edu/goldenpath/help/bigWig.html) format is in an indexed binary format useful for displaying dense, continuous data in Genome Browsers such as the [UCSC](https://genome.ucsc.edu/cgi-bin/hgTracks) and [IGV](http://software.broadinstitute.org/software/igv/). This mitigates the need to load the much larger BAM files for data visualisation purposes which will be slower and result in memory issues. The bigWig format is also supported by various bioinformatics software for downstream processing such as meta-profile plotting.
+_Description_:
+The [bigWig](https://genome.ucsc.edu/goldenpath/help/bigWig.html) format is in an indexed binary format useful for displaying dense, continuous data in Genome Browsers such as the [UCSC](https://genome.ucsc.edu/cgi-bin/hgTracks) and [IGV](http://software.broadinstitute.org/software/igv/). This mitigates the need to load the much larger BAM file for data visualisation purposes which will be slower and result in memory issues. The bigWig format is also supported by various bioinformatics software for downstream processing such as meta-profile plotting.
```
[the remark this answers is not in the document]
Comment (yuukiiwa) [reply docs/output.md:134]: I have updated this

File: docs/output.md [hunk 26]
```diff
@@ -89,68 +80,106 @@ If you would like to run NanoLyse on the raw FastQ files then you can provide `-
...
+  - `variant_calling/sniffles/<SAMPLE>.vcf.gz.tbi`: index for zipped VCF file with structural variants.
+
+</details>
+
+_Documentation_:
+[Medaka](https://github.com/nanoporetech/medaka), [DeepVariant](https://github.com/google/deepvariant), [PEPPER-Margin-DeepVariant](https://github.com/kishwarshafin/pepper),
+[Sniffles](https://github.com/fritzsedlazeck/Sniffles), [cuteSV](https://github.com/tjiangHIT/cuteSV)
+
+_Description_:
+If the `--protocol DNA` and the `--call_variants` parameters are defined small and structural variants.
```
[the remark this answers is not in the document]
Comment (yuukiiwa) [reply docs/output.md:179]: I have updated this

File: docs/output.md [hunk 27]
```diff
@@ -159,86 +188,114 @@ The creation of these files can be bypassed by setting the parameters `--skip_bi
 
 If bambu is used:
 
-* `bambu/`
-    * `extended_annotations.gtf` - a gtf file that contains both annotated and novel transcripts
-    * `counts_gene.txt` - gene expression estimates
-    * `counts_transcript.txt` - transcript expression estimates
+- `bambu/`
+  - `extended_annotations.gtf`: a GTF file that contains both annotated and novel transcripts
```
[the remark this answers is not in the document]
Comment (yuukiiwa) [reply docs/output.md:192]: I have updated this

File: docs/output.md [hunk 28]
```diff
@@ -159,86 +188,114 @@ The creation of these files can be bypassed by setting the parameters `--skip_bi
...
+  - `counts_gene.txt`: a TXT file containing gene expression estimates
```
[the remark this answers is not in the document]
Comment (yuukiiwa) [reply docs/output.md:193]: I have updated this

File: docs/output.md [hunk 29]
```diff
@@ -159,86 +188,114 @@ The creation of these files can be bypassed by setting the parameters `--skip_bi
...
+  - `counts_transcript.txt`: a TXT file containing transcript expression estimates
```
[the remark this answers is not in the document]
Comment (yuukiiwa) [reply docs/output.md:194]: I have updated this

File: conf/test_nobc_nodx_vc.config [hunk 30]
```diff
@@ -0,0 +1,29 @@
...
+    // Input data to skip basecalling and demultiplexing, and variant call
+    input               = 'https://raw.githubusercontent.com/nf-core/test-datasets/nanoseq/3.0/samplesheet/samplesheet_nobc_nodx_vc.csv'
+    protocol            = 'DNA'
+    skip_basecalling    = true
+    skip_quantification = true
+    skip_demultiplexing = true
+    call_variants       = true
+
+    variant_caller      = 'deepvariant'
+    structural_variant_caller = 'cutesv'
```
[the remark this answers is not in the document]
Comment (yuukiiwa) [reply conf/test_nobc_nodx_vc.config:28]: I have updated this

File: .github/workflows/ci.yml (hunk 18 above)
[the remark this answers is not in the document]
Comment (christopher-hakkaart) [reply .github/workflows/ci.yml:64]: I'm an idiot

File: modules/local/m6anet_dataprep.nf [hunk 31]
```diff
@@ -0,0 +1,27 @@
+process M6ANET_DATAPREP {
+    tag "$meta.id"
+    label 'process_medium'
+
+//  conda     (params.enable_conda ? "bioconda::nanopolish==0.13.2" : null) // need to get xpore onto conda
```
[the remark this answers is not in the document]
Comment (christopher-hakkaart) [reply modules/local/m6anet_dataprep.nf:5]: Ignore this one

[the remark this answers is not in the document]
Comment (yuukiiwa) [reply modules/local/m6anet_dataprep.nf:5]: `m6anet` is not on `conda` yet

**Review by christopher-hakkaart — APPROVED** [review]

**Pull request merged**